clippy_lints/src/eq_op.rs

```diff
@@ -52,9 +60,34 @@ declare_clippy_lint! {
 
 declare_lint_pass!(EqOp => [EQ_OP, OP_REF]);
 
+const ASSERT_MACRO_NAMES: [&str; 4] = ["assert_eq", "assert_ne", "debug_assert_eq", "debug_assert_ne"];
+
 impl<'tcx> LateLintPass<'tcx> for EqOp {
     #[allow(clippy::similar_names, clippy::too_many_lines)]
     fn check_expr(&mut self, cx: &LateContext<'tcx>, e: &'tcx Expr<'_>) {
+        if let ExprKind::Block(ref block, _) = e.kind {
+            for stmt in block.stmts {
+                for amn in &ASSERT_MACRO_NAMES {
+                    if_chain! {
+                        if is_expn_of(stmt.span, amn).is_some();
+                        if let StmtKind::Semi(ref matchexpr) = stmt.kind;
```

Comment:
Ideally we would not have to know the structure of the expanded code at this point (just the `is_expn_of` call maybe), I think that this should be handled by `extract_assert_macro_args` if possible.

Reply:
The `assert_*` macros are not expanded the same way as `debug_assert*` macros are. There is one `Block` level more in the latter (due to the test of the variable enabling the debug output I suppose). So if we have multiple `assert_eq!` calls in the same function, when doing the `check_expr` these macros' `Expr` are only accessible under `block.stmts`.

IMO if we really want `extract_assert_macro_args` to handle these lines, we will have to be able to process multiple macros inside. It will make the return type and the reporting of the macros parsed (comparing/storing its name for linting) more complex to manage.

Reply:
Alright, let's leave it like this then. We can change it in a followup if we find a simpler way to do it.